### PR TITLE
[NR-57248] fix(dependabot): check autor in commit

### PR DESCRIPTION
**Pull request by paologallinaharbur**
Fix https://github.com/newrelic/release-toolkit/issues/108

I am keeping both checks, author, and commit message in case there is no info that can be used

### CODE REVIEW

File: src/changelog/sources/dependabot/source_test.go [hunk 1]
```diff
@@ -40,134 +40,140 @@ func TestSource_Source(t *testing.T) {
 		expected []changelog.Dependency
 	}{
 		{
-			commit: git.Commit{Message: "Whatever actions/github-script from 1.0.2 to 1.0.4."},
+			commit: git.Commit{Message: "Whatever actions/github-script from 1.0.2 to 1.0.4.", Author: "dependabot[bot]"},
```
Comment (roobre) [inline src/changelog/sources/dependabot/source_test.go:43]: To make the test readable, we can "cheat" a bit and, in the test body, set `Author` to a default value if it's empty.

File: src/changelog/sources/renovate/source.go [hunk 2]
```diff
@@ -47,8 +47,6 @@ func (r Source) Changelog() (*changelog.Changelog, error) {
 		var commitDependencies []changelog.Dependency
 
 		commitLine := strings.Split(c.Message, "\n")[0]
-		// Contrary to dependabot, where we can reliably tell if a message comes from it by just inspecting the message,
```
Comment (roobre) [inline src/changelog/sources/renovate/source.go:50]: > where we can reliably tell

Ominous foreshadowing 🤣

**Review by roobre — APPROVED** [review]
LGTM, left a small suggestion regarding tests

File: src/changelog/sources/dependabot/source_test.go (hunk 1 above)
Comment (paologallinaharbur) [reply src/changelog/sources/dependabot/source_test.go]: done

**Pull request merged**